src/main/scala/de/frosner/ddq/core/Check.scala

```diff
@@ -563,6 +573,41 @@ object Check {
   )
 
   /**
+   * Check whether columns in the dependent set have functional dependency on determinant set.
+   * @param determinantSet sequence of column names which form a determinant set
+   * @param dependentSet sequence of column names which form a dependent set
+   * @return [[core.Constraint]] object
+   */
+  def hasFunctionalDependency(determinantSet: Seq[String], dependentSet: Seq[String]) = Constraint(
+    df => {
+      val determinantIndexes = df.columns.zipWithIndex.filter {
+        case (name, index) => determinantSet.contains(name)
+      }.map { case (name, index) => index }
+      val dependentIndexes = df.columns.zipWithIndex.filter {
+        case (name, index) => dependentSet.contains(name)
+      }.map { case (name, index) => index }
+      val rdd = df.map(row => (
```

Comment:
Please try to use more speaking variable names. Consider splitting multiple transformations if this helps to make it more readable.